### PR TITLE
[fix](function)Substring input UTF-8 character error result

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes

The error occurs because the length is calculated using the byte length, not the UTF-8 length.
```
mysql> select SUBSTRING('中文测试',5);
+------------------------------------------+
| substring('中文测试', 5, 2147483647)     |
+------------------------------------------+
| 中文测试                                 |
+------------------------------------------+
```

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

File: regression-test/suites/query_p0/sql_functions/string_functions/test_string_function.groovy [hunk 1]
```diff
@@ -228,6 +228,9 @@ suite("test_string_function", "arrow_flight_sql") {
     qt_sql "select substring('abcdef',3,-1);"
     qt_sql "select substring('abcdef',-3,-1);"
     qt_sql "select substring('abcdef',10,1);"
+    qt_sql "select substring('中文测试',5);"
+    qt_sql "select substring('中文测试',4);"
+    qt_sql "select substring('中文测试',2,2);"
```
Comment (yiguolei) [inline regression-test/suites/query_p0/sql_functions/string_functions/test_string_function.groovy:234]: add test for negative parameters and Chinese chars

**Comment by Mryange** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]

**Review by yiguolei — CHANGES_REQUESTED** [review]
need some discussion

**Pull request closed without merging**